ledger-core/v2/virtual/handlers/vdelegatedcallrequest.go

```diff
@@ -0,0 +1,204 @@
+
+		pendingTableFull bool
+	)
+
+	action := func(state *object.SharedState) {
+		var (
+			oldestPulse  pulse.Number
+			pendingList  *object.RequestList
+			pendingCount uint8
+		)
+
+		callTolerance := s.Payload.CallFlags.GetInterference()
+
+		switch callTolerance {
+		case contract.CallTolerable:
+			pendingList = state.PendingTable.GetList(contract.CallTolerable)
+			oldestPulse = state.OrderedPendingEarliestPulse
+			pendingCount = state.ActiveOrderedPendingCount
+		case contract.CallIntolerable:
+			pendingList = state.PendingTable.GetList(contract.CallIntolerable)
+			oldestPulse = state.UnorderedPendingEarliestPulse
+			pendingCount = state.ActiveUnorderedPendingCount
+		default:
+			panic(throw.Unsupported())
+		}
+
+		if oldestPulse == pulse.Unknown || s.Payload.RequestReference.GetLocal().GetPulseNumber() < oldestPulse {
+			tooOldRequest = true
+			return
+		}
+
+		// pendingList already full
+		if pendingList.Count() == int(pendingCount) && !pendingList.Exist(s.Payload.RequestReference) {
+			pendingTableFull = true
+			return
+		}
+
+		if !pendingList.Add(s.Payload.RequestReference) {
+			ctx.Log().Warn("request already processed")
+			// request was already in the list we will delegate token, maybe it is repeated call
+			return
+		}
+
+		if pendingList.Count() == int(pendingCount) {
+			state.SetPendingListFilled(ctx, callTolerance)
+		}
+	}
+
+	switch s.objectSharedState.Prepare(action).TryUse(ctx).GetDecision() {
+	case smachine.Passed:
+	case smachine.NotPassed:
+		return ctx.WaitShared(s.objectSharedState.SharedDataLink).ThenRepeat()
+	default:
+		panic(throw.Impossible())
+	}
+
+	if tooOldRequest {
+		ctx.Log().Warn(struct {
+			*log.Msg  `txt:"There is no such object"`
+			Reference string
+		}{Reference: s.Payload.RequestReference.String()})
+		return ctx.Error(throw.IllegalValue())
+	}
+
+	if pendingTableFull {
+		ctx.Log().Warn(struct {
+			*log.Msg  `txt:"Pending table already full"`
+			Reference string
+		}{Reference: s.Payload.RequestReference.String()})
+		return ctx.Error(throw.IllegalState())
+	}
+
+	return ctx.Jump(s.buildResponse)
+}
+
+func (s *SMVDelegatedCallRequest) buildResponse(ctx smachine.ExecutionContext) smachine.StateUpdate {
+	target := s.Meta.Sender
+
+	response := payload.VDelegatedCallResponse{
+		RefIn: s.Payload.RefIn,
+		DelegationSpec: payload.CallDelegationToken{
+			TokenTypeAndFlags: payload.DelegationTokenTypeCall,
+			Approver:          s.jetCoordinator.Me(),
+			DelegateTo:        s.Meta.Sender,
+			PulseNumber:       s.pulseSlot.PulseData().PulseNumber,
+			Callee:            s.Payload.Callee,
+			Caller:            s.Meta.Sender,
+			ApproverSignature: []byte{0xde, 0xad, 0xbe, 0xef},
+		},
+	}
+
+	s.messageSender.PrepareAsync(ctx, func(goCtx context.Context, svc messagesender.Service) smachine.AsyncResultFunc {
+		err := svc.SendTarget(goCtx, &response, target)
+		return func(ctx smachine.AsyncResultContext) {
+			if err != nil {
+				ctx.Log().Error("failed to send message", err)
```

Comment:
@sexyhamster Do we have some information about which message send failed? (Maybe error captured stack or something else, I don't know)

Reply:
added more information to log

Reply:
as I remember it correctly, Log gives us SM and step it was called on.

Reply:
but it lacks of information about which request/object is processed, so I've added more info here